plugins/modules/vm_snapshot.py

```diff
@@ -145,6 +150,31 @@
 from ..module_utils.vm import VM
 
 
+def get_snapshot(
```

Comment:
Any chance to get a snapshot with correct label, but from a different VM?

It seems to me: filter to `domainUUID=vm_object.uuid` part is lost. And integration tests should be extended too.

Reply:
We pass VMobject as one of the parameters in this function.
It is then used with label for snapshot query if snapshot_uuid wasn't passed as parameter to Ansible.
 The only thing I added to this part was the `if module.params["uuid"]:` , everything else is the same.
 UUID is unique throughout the platform and use a different endpoint.